### PR TITLE
Kafka based publisher for replication tasks

### PR DESCRIPTION
**Pull request by samarabbas**

### CODE REVIEW

File: common/messaging/kafkaClient.go [hunk 1]
```diff
@@ -65,3 +68,16 @@ func (c *kafkaClient) NewConsumer(topicName, consumerName string, concurrency in
 	consumer, err := c.client.NewConsumer(consumerConfig)
```
Comment (madhuravi) [inline common/messaging/kafkaClient.go]: Fixing the interface issue with the latest kafka client as part of this?

File: common/messaging/kafkaClient.go [hunk 2]
```diff
@@ -30,6 +32,7 @@ type (
 	kafkaClient struct {
 		config *KafkaConfig
 		client *kafkaclient.Client
```
Comment (madhuravi) [inline common/messaging/kafkaClient.go]: I think all you need is to glide up and make this a non-pointer

File: common/messaging/kafkaClient.go (hunk 1 above)
Comment (samarabbas) [reply common/messaging/kafkaClient.go]: I will take care of it in a separate PR.

File: common/messaging/kafkaClient.go (hunk 2 above)
Comment (samarabbas) [reply common/messaging/kafkaClient.go]: I will take care of it in a separate PR.

**Pull request merged**